From 8bdede57903f53677095e2062363c08d8b2a386a Mon Sep 17 00:00:00 2001
From: Nick Pelly <npelly@google.com>
Date: Mon, 23 Mar 2009 18:17:16 -0700
Subject: [PATCH 043/247] Bluetooth: Fallback from eSCO to SCO on error code 0x1f (unspecified error).

Kyocera ED-8800 headset returns this error code when eSCO is attempted.

Signed-off-by: Nick Pelly <npelly@google.com>
---
 net/bluetooth/hci_event.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/net/bluetooth/hci_event.c b/net/bluetooth/hci_event.c
index 5553424..42cb717 100644
--- a/net/bluetooth/hci_event.c
+++ b/net/bluetooth/hci_event.c
@@ -1646,7 +1646,8 @@ static inline void hci_sync_conn_complete_evt(struct hci_dev *hdev, struct sk_bu
 		conn->type = SCO_LINK;
 	}
 
-	if (conn->out && ev->status == 0x1c && conn->attempt < 2) {
+	if (conn->out && (ev->status == 0x1c || ev->status == 0x1f) &&
+			conn->attempt < 2) {
 		conn->pkt_type = (hdev->esco_type & SCO_ESCO_MASK) |
 					(hdev->esco_type & EDR_ESCO_MASK);
 		hci_setup_sync(conn, conn->link->handle);
-- 
1.7.0.4

